### PR TITLE
Add Hyphen Handling

### PR DESCRIPTION
**Pull request by kx-chen**
Add tests and new regex to handle leading, trailing, and consecutive hyphens in username. 
Checked it over and it seems to be good (it's late here so i probably made some mistakes, please advise).

Closes https://github.com/andrewda/valid-github-username/issues/1

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module.exports = (name) => {
-  const replaceRegex = /[^a-zA-Z0-9-]/g;
-  const replaced = name.replace(replaceRegex, '');
+  const replaceRegex = /[^a-zA-Z0-9-]*(-{2,}?)*(-$)*(^-)*/g;
+  let replaced = name.replace(replaceRegex, '');
```
Comment (wisn) [inline index.js]: Could you elaborate why replace `const` to `let`? I didn't see that `replaced` is modified afterwards. Better to use `const`.

```suggestion
  const replaced = name.replace(replaceRegex, '');
```

File: test.js [hunk 2]
```diff
@@ -13,6 +13,22 @@ describe('removes illegal characters', () => {
    });
 });
 
+describe('removes illegal hyphen placement',() => {
```
Comment (wisn) [inline test.js]: This is trivial but please add a space after comma.

```suggestion
describe('removes illegal hyphen placement', () => {
```

**Review by wisn — CHANGES_REQUESTED** [review]

**Review by andrewda — CHANGES_REQUESTED** [review]
Looks good other than changes @wisn suggested. Also, please edit title to something imperative tense, like "Add hyphen handling". That will just make it easier for me to squash and merge, and make the PR easier to find in the git history 😄

**Comment by kx-chen** [comment]
Amended. Changing the `const` to `let` was unintended. Whoops.

File: test.js [hunk 3]
```diff
@@ -13,6 +13,22 @@ describe('removes illegal characters', () => {
...
+    it('removes trailing dashes', () => {
+        let username = validGitHubUsername('kxchen-');
+        assert.equal(username, 'kxchen');
+    });
+
+    it('removes leading dashes', () => {
+        let username = validGitHubUsername('-kxchen');
+        assert.equal(username, 'kxchen');
+    });
+});
```
Comment (andrewda) [inline test.js:31]: Add a newline after this line

**Review by andrewda — CHANGES_REQUESTED** [review]
After that it looks good!

File: test.js (hunk 3 above)
Comment (kx-chen) [reply test.js:31]: Should be resolved.

**Review by wisn — APPROVED** [review]

**Review by andrewda — APPROVED** [review]

**Pull request merged**